File: server/Netptune.Core/ViewModels/Notifications/NotificationViewModel.cs
using System.Text.Json.Serialization;

using Netptune.Core.Enums;

namespace Netptune.Core.ViewModels.Notifications;

public class NotificationViewModel
{
    public int Id { get; set; }

    public bool IsRead { get; set; }

    public string? Link { get; set; }

    public EntityType EntityType { get; set; }

    public ActivityType ActivityType { get; set; }

    public DateTime CreatedAt { get; set; }

    public string ActorUserId { get; set; } = null!;

    public string ActorUsername { get; set; } = null!;

    public string? ActorPictureUrl { get; set; }

    public bool ActorIsServiceAccount { get; set; }

    public string? EntityName { get; set; }

    public string? EntityIdentifier { get; set; }

    public int? ActivityEntryId { get; set; }

    public int? RevisionCount { get; set; }

    [JsonIgnore]
    public string[]? ChangedFieldsArray { get; set; }

    public List<string> ChangedFields => ChangedFieldsArray is null ? [] : [.. ChangedFieldsArray];
}

File: client/src/app/core/models/view-models/notification-view-model.ts
import { ActivityType } from './activity-view-model';
import { EntityType } from '../entity-type';

export interface NotificationViewModel {
  id: number;
  isRead: boolean;
  link: string | null;
  entityType: EntityType;
  activityType: ActivityType;
  createdAt: Date;
  actorUserId: string;
  actorUsername: string;
  actorPictureUrl: string | null;
  entityName: string | null;
  entityIdentifier: string | null;
}
